src/resources/swagger/index.yaml

```
@@ -88,6 +88,35 @@ paths:
           description: The error message indicates something wrong with the request
         '500':
           description: Internal server error
+
+  '/clinical/api/donors/data-for-all-files':
+    parameters:
+      - name: filter
+        description: JSON filter following the SQON spec
+        in: query
+        required: false
+        schema:
+          type: string
+    post:
+      tags:
+        - Clinical API
+      security:
+        - bearerAuth: []
+      summary: Download clinical data for all available files from filter.
+      description: Download clinical data for all available files from filter.
```

thinking this may be a valid nuance?
```suggestion
      summary: Download clinical data for all available files from (optional) filters.
      description: Download clinical data for all available files from (optional) filters.
```

---

thinking further on this, the "all" part of this endpoint implies to me no filtering should be possible. I mean why would a query give you partial results, unless it is due to pagination or hard limits?

---

I would rephrase this a bit farther even:

```
Download clinical data associated with the donors for all files. A SQON filter can be provided to restrict the data retrieved to the files returned by a specific query.
```